libyul/AsmAnalysis.cpp

```diff
@@ -302,7 +306,22 @@ vector<YulString> AsmAnalyzer::operator()(FunctionCall const& _funCall)
 	{
 		Expression const& arg = _funCall.arguments[i - 1];
 
-		argTypes.emplace_back(expectExpression(arg));
+		yulAssert(!m_visitingLiteralUsedAsLiteralArgument, "");
+		m_visitingLiteralUsedAsLiteralArgument =
+			needsLiteralArguments &&
+			(*needsLiteralArguments)[i - 1] &&
+			holds_alternative<Literal>(arg);
+
+		try
+		{
+			argTypes.emplace_back(expectExpression(arg));
+			m_visitingLiteralUsedAsLiteralArgument = false;
+		}
+		catch (...)
```

Comment:
We usually don't do cleanup after an exception because it always means that we stop doing what we were about to do anyway.

Reply:
`try` block removed.